entity-management-web/src/main/java/eu/europeana/entitymanagement/web/EMController.java

```diff
@@ -96,6 +97,69 @@
 
     }
 
+    @ApiOperation(value = "Update an entity", nickname = "updateEntity", response = java.lang.Void.class)
+    @RequestMapping(value = { "/{type}/base/{identifier}", "/{type}/{identifier}" },method = RequestMethod.PUT, produces = {
+	    HttpHeaders.CONTENT_TYPE_JSONLD, MediaType.APPLICATION_JSON_VALUE})
+    public ResponseEntity<EntityRecord> updateEntity(
+    	@RequestHeader(value = "If-Match", required = false) String ifMatchHeader,
+	    @RequestParam(value = CommonApiConstants.PARAM_WSKEY, required = false) String wskey,
+	    @RequestParam(value = WebEntityConstants.QUERY_PARAM_PROFILE, required = true, defaultValue = "external") String profile,
+	    @PathVariable(value = WebEntityConstants.PATH_PARAM_TYPE) String type,
+	    @PathVariable(value = WebEntityConstants.PATH_PARAM_IDENTIFIER) String identifier,
+	    @RequestBody EntityPreview entityCreationRequest,
+	    HttpServletRequest request) {
+
+    	// TODO: Re-enable authentication
+    	// verifyReadAccess(request);
+
+    	EntityRecord existingRecord = entityRecordRepository.findByEntityId(getEntityUri(type, identifier));
+    	if(existingRecord!=null) {
+
+    		Date timestamp = (existingRecord.getEntity().getIsAggregatedBy() != null) ? existingRecord.getEntity().getIsAggregatedBy().getModified() : null;
+			Date etagDate = (timestamp != null)? timestamp : new Date();
+			String etag = generateETag(etagDate, FormatTypes.jsonld.name(), getApiVersion());
+
+			try {
+				checkIfMatchHeader(etag, request);
+			} catch (HttpException e) {
+				return ResponseEntity.status(HttpStatus.PRECONDITION_FAILED).header("info:", "The value of the If-Match HTTP header does not allign with the given ETag value generated from the timestamp.").build();
+			}
+
+			//TODO: update the Europeana proxy and also call the UpdateTask for updating the entity if it is meant to be used for that
+			if(entityCreationRequest.getId()!=null) {
+				existingRecord.getEntity().setEntityId(entityCreationRequest.getId());
```

Comment:
existingRecord.getEntity() is the consolidated version, not the europeana proxy.  All these following operations must be performed on the Europeana Proxy, which must be retried from the EntityRecordImpl.proxies

Reply:
Done.